contracts/utils/Create2Factory.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/**
+ * @dev This contract is used to deploy smart contracts with create2 op code.
+ * in introduced at https://github.com/Conflux-Chain/CIPs/blob/master/CIPs/cip-31.md
+ * is deployed at Conflux's genesis block, it's address is 0x8A3A92281Df6497105513B18543fd3B60c778E40
+ */
+contract Create2Factory {
+    mapping(address => bool) deployed;
+
+    /**
+     * @dev Check whether a address is deployed with Create2Factory
+     * @param addr The address to check
+     */
+    function isDeployed(address addr) public view returns (bool) {
+        return deployed[addr];
+    }
+
+    /**
+     * @dev get current Sponsored Balance for gas
+     * @param code The code to depoly
+     * @param salt The salt used to generate contract address
+     */
+    function deploy(bytes memory code, uint256 salt) public returns (address) {
+        address addr;
+        bool success = true;
+        assembly {
+            addr := create2(0, add(code, 0x20), mload(code), salt)
+            if iszero(extcodesize(addr)) {
+                success := 0
+            }
+        }
+        require(success, "create2 failed");
+        require(!deployed[addr], "contract has been deployed before");
```

Comment:
Shouldn't this be checked before calling `create2`? (This is just an observation, I guess the contract cannot be changed now).

Reply:
Yes, you are right. And the code cannot be changed now